File: 01-arrays-and-hashing/0217-contains-duplicate.py
"""
LeetCode 217 · Contains Duplicate · Easy
https://leetcode.com/problems/contains-duplicate/

Pattern : Hash Map (frequency)
Solved  : 14 Aug 2026
Time    : O(n)
Space   : O(n)
"""
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        count = {}
        for num in nums:
            if num in count:
                count[num] += 1
                return True
            else:
                count[num] = 1
        return False

File: 2025/12-December/24-12-25/0217-contains-duplicate.cpp
/*
 * LeetCode 217 · Contains Duplicate · Easy
 * https://leetcode.com/problems/contains-duplicate/
 *
 * Pattern : Hash Set
 * Solved  : 24 Dec 2025
 * Time    : O(n)
 * Space   : O(n)
 */

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> st; // Unordered Set
        for(auto &x :nums){
            if(st.count(x)) { // Duplicate Found
                return true ;
            }
            st.insert(x); // Duplicate Not Found! Inserting in Set
        }
        return false;
    }
};
